### ISSUE TITLE
[rllib] Tame the configuration

### ISSUE THREAD

**Issue by drozzy** [ray-project/ray#8571]
Hi, guys. 
Is there any way to make configuration easier? Or more manageable...

The present config system suffers from two major issues for me personally:

## 1. Wall of text

There is just so much of it that it's just hard to know what's more important and what's not. Comments are intermixed with code and it's hard to scan through it. I find myself going through the tuned examples to get a better feel of what's important and get a feel for good starting values. However, even after using rllib for a bit, I find it very hard to read the config sections:

<img width="667" alt="Screen Shot 2020-05-23 at 2 57 21 AM" src="https://user-images.githubusercontent.com/140710/82724040-21441880-9ca1-11ea-8497-d6b74cea0300.png">

## 2. Configuration inheritance 

Configuration inheritance makes it hard to know which config to find where. 

For example, if we look at [PPO-specific configs](https://docs.ray.io/en/master/rllib-algorithms.html#proximal-policy-optimization-ppo), the learning rate `lr` parameter is overriden from base config here but `kl_coeff` is completely new parameter. But looking at PPO config the distinction is not apparent. 

As such, I find myself often jumping between pages and trying to remember where which config value was defined.

**Comment by drozzy** [comment]
Possible solution would be to get rid of config dicts and replace them with named parameters. 

Look at pytorch. The usage is clear and beautiful at the expense of may parameters. 

But that expense is paid by the framework developer. Whereas dict config makes it easy for framework developer but hard for the end user.

**Comment by drozzy** [comment]
Additionally the common config creates a false sense of hierarchy. 
The common config is not a config that is relevant to PPO, for example. It is a rllib config. 

As such, it might make more sense to pass  an instance of rllib to the instance of PPO algorithm:

    rllib = RLLib(numeorkers=2)
    ppo = PPO(rllib, lr=0.01)

This is a rough sketch. But this makes it way clearer what goes where.
 
This might also simply the ppo trainer and ray invocations which presently take a bunch of dicts as well. Named parameters are much better as that is what they are there for. 

Config should be immediately converted to named parameters as soon as it is parsed. 
Instances of classes should be injected into other classes as needed.

**Comment by drozzy** [comment]
Example of named parameters config:

![C17A7B6E-915D-4126-B7C7-E55C835EE398](https://user-images.githubusercontent.com/140710/84259579-b0955c80-aae6-11ea-8ddd-7951c0c2eb84.jpeg)
